packages/upset/src/components/AltTextSidebar.tsx

```diff
@@ -75,65 +65,42 @@ export const AltTextSidebar: FC<Props> = ({ open, close, generateAltText }) => {
         setTextGenErr(false);
       } catch (e) {
         const msg: string = (e as Error).message;
-        // We want the error message to display on the frontend
-        setAltText({
-          longDescription: msg,
-          shortDescription: msg,
-          techniqueDescription: msg,
-        });
-        setTextGenErr(true);
+        setTextGenErr(msg);
       }
     }
-
+    
     generate();
   }, [currState]);
 
-  // this useEffect resets the plot information when the edit is toggled off
-  useEffect(() => {
-    if (!isEditable) {
-      setPlotInformation(plotInformationState);
+  // Current alt text to display to the user
+  const displayAltText: string | undefined = useMemo(() => {
+    if (textEditing) {
+      return useLong
+      ? userLongText ?? altText?.longDescription
+      : userShortText ?? altText?.shortDescription
     }
-  }, [isEditable]);
-
-  // this useEffect sets the plot information state to match the trrack state
-  useEffect(() => {
-    // this will prevent the state from being reset while the user is editing the form values
-    if (!isEditable) {
-      setPlotInformation(plotInformationState);
-    }
-  });
-
-  const generatePlotInformationText = () => {
-    // return default string if there are no values filled in
-    if (Object.values(plotInformation).filter((a) => a.length > 0).length === 0) {
-      return `This UpSet plot shows ${placeholderText.description}. The sets are ${placeholderText.sets}. The items are ${placeholderText.items}`;
-    }
-
-    let str: string = '';
-    if (plotInformation.description !== '') {
-      str += `This UpSet plot shows ${plotInformation.description}. `;
-    }
-    if (plotInformation.sets !== '') {
-      str += `The sets are ${plotInformation.sets}. `;
-    }
-    if (plotInformation.items !== '') {
-      str += `The items are ${plotInformation.items}.`;
-    }
-
-    return str;
-  };
-
-  const handleEditableChange = () => {
-    setIsEditable(!isEditable);
-  };
-
+    return useLong 
+    ? currState.useUserAlt ? userLongText : altText?.longDescription 
+    : currState.useUserAlt ? userShortText : altText?.shortDescription;
+  }, [useLong, userLongText, userShortText, altText, currState.useUserAlt]);
```

Comment:
Are you able to specifiy `altText.shortDescription` and `altText.longDescription` to avoid the entire object as dependency?